source/core/TypeAndDataManager.cpp

```diff
@@ -778,14 +786,22 @@ Boolean TypeCommon::IsA(TypeRef otherType, Boolean compatibleStructure)
                 otherEncoding == kEncoding_Ascii || otherEncoding == kEncoding_Unicode) {
                 bMatch = TopAQSize() == otherType->TopAQSize();
             }
-        } else if (thisEncoding == kEncoding_Cluster && otherEncoding == kEncoding_Cluster &&
-                   !this->IsIntrinsicClusterDataType() && !otherType->IsIntrinsicClusterDataType()) {
-            if (this->SubElementCount() == otherType->SubElementCount()) {
-                for (Int32 i = 0; i < this->SubElementCount(); i++) {
-                    if (!this->GetSubElement(i)->CompareType(otherType->GetSubElement(i)))
-                        break;
+        } else if (thisEncoding == kEncoding_Cluster && otherEncoding == kEncoding_Cluster) {
```

Comment:
Ok with me, both functions seem to be similar in nature, but have subtle differences and are used in several places. It is a good idea to do the refactoring in another changeset.